components/view_image/view_image.jsx

```diff
@@ -312,9 +346,30 @@ export default class ViewImageModal extends React.PureComponent {
             );
         }
 
-        let closeButtonClass = 'modal-close';
+        let closeButton;
         if (this.state.showCloseBtn) {
-            closeButtonClass += ' modal-close--show';
+            closeButton = (
+                <div
+                    className='modal-close'
+                    onClick={this.props.onModalDismissed}
+                />
+            );
+        }
+
+        let zoomButtons;
+        if (this.state.showZoomBtn) {
+            zoomButtons = (
+                <div>
+                    <div
+                        className='modal-zoom-in'
+                        onClick={this.handleZoomIn}
```

Comment:
Perhaps a possible solution is using `debounce` from lodash to set a min time between clicks to call the function as well as a max wait time to call. Something like:

`onClick={debounce(this.handleZoomIn, 300, {maxWait: 300})}`

However, the errors I mentioned in [my comment]() still exists for larger file sized PDFs, so I'm not sure if this is the best solution.